docs/testing-functions.md

```diff
@@ -0,0 +1,198 @@
+  // req and res are strongly typed here
+});
+```
+
+The testing module provides a `getFunction` helper method that can be used to
+access a function that was registered with the Functions Framework. To use it in
+your unit test you must first load the module that registers the function you wish
+to test.
+
+```js
+import {getFunction} from "@google-cloud/functions-framework/testing";
+
+describe("HelloTests", () => {
+  before(async () => {
+    // load the module that defines HelloTest
+    await import("./hello_tests.js");
+  });
+
+  it("is testable", () => {
+    // get the function using the name it was registered with
+    const HelloTest = getFunction("HelloTests");
+    // ...
+  });
+});
+```
+
+## Testing HTTP functions
+
+Testing an HTTP function is generally as simple as generating a request, calling
+the function, and asserting against the response.
+
+HTTP functions are passed an [express.Request](https://expressjs.com/en/api.html#req)
+and an [express.Response](https://expressjs.com/en/api.html#res) as arguments. You can
+create simple stubs to use in unit tests.
+
+```js
+import assert from "assert";
+import {getFunction} from "@google-cloud/functions-framework/testing";
+
+describe("HelloTests", () => {
+  before(async () => {
+    // load the module that defines HelloTest
+    await import("./hello_tests.js");
+  });
+
+  it("is testable", () => {
+    // get the function using the name it was registered with
+    const HelloTest = getFunction("HelloTests");
+
+    // a Request stub with a simple JSON payload
+    const req = {
+      body: { foo: "bar" },
+    };
+    // a Response stub that captures the sent response
+    let result;
+    const res = {
+      send: (x) => {
+        result = x;
+      },
+    };
+
+    // invoke the function
+    HelloTest(req, res);
+
+    // assert the response matches the expected value
+    assert.equal(result, "Hello, World!");
+  });
+});
+```
+
+## Testing CloudEvent functions
+
+Testing a CloudEvent function works similarly. The `testing` module provides
+methods to help construct example CloudEvent objects that can be used to invoke
+the function.
+
+Unlike HTTP functions, event functions do not accept a response argument. Instead, you
+will need to test side effects. A common approach is to replace your function's
+dependencies with mock objects that can be used to verify its behavior. The
+[sinonjs](https://sinonjs.org/) is a standalone library for creating mocks that work with
+any Javascript testing framework:
+
+```js
+import assert from "assert";
+import sinon from "sinon";
+import {
+  getFunction,
+  testCloudEvent,
+} from "@google-cloud/functions-framework/testing";
+
+import {MyDependency} from "./my_dependency.js";
+
+describe("HelloCloudEvent", () => {
+  before(async () => {
+    // load the module that defines HelloCloudEvent
+    await import("./hello_cloud_event.js");
+  });
+
+  const sandbox = sinon.createSandbox();
+
+  beforeEach(function () {
```

Comment:
nit: Use arrow functions consistently throughout this project / file. `beforeEach(() => {`

Here and elsewhere.